tests/collapse-perf-tests.rs

```diff
@@ -0,0 +1,222 @@
+    perf_cycles_instructions_01_addrs,
+
+    perf_dd_stacks_01_pid,
+    perf_dd_stacks_01_tid,
+    perf_dd_stacks_01_kernel,
+    perf_dd_stacks_01_jit,
+    perf_dd_stacks_01_all,
+    perf_dd_stacks_01_addrs,
+
+    perf_funcab_cmd_01_pid,
+    perf_funcab_cmd_01_tid,
+    perf_funcab_cmd_01_kernel,
+    perf_funcab_cmd_01_jit,
+    perf_funcab_cmd_01_all,
+    perf_funcab_cmd_01_addrs,
+
+    perf_funcab_pid_01_pid,
+    perf_funcab_pid_01_tid,
+    perf_funcab_pid_01_kernel,
+    perf_funcab_pid_01_jit,
+    perf_funcab_pid_01_all,
+    perf_funcab_pid_01_addrs,
+
+    perf_iperf_stacks_pidtid_01_pid,
+    perf_iperf_stacks_pidtid_01_tid,
+    perf_iperf_stacks_pidtid_01_kernel,
+    perf_iperf_stacks_pidtid_01_jit,
+    perf_iperf_stacks_pidtid_01_all,
+    perf_iperf_stacks_pidtid_01_addrs,
+
+    perf_java_faults_01_pid,
+    perf_java_faults_01_tid,
+    perf_java_faults_01_kernel,
+    perf_java_faults_01_jit,
+    perf_java_faults_01_all,
+    perf_java_faults_01_addrs,
+
+    perf_java_stacks_01_pid,
+    perf_java_stacks_01_tid,
+    perf_java_stacks_01_kernel,
+    perf_java_stacks_01_jit,
+    perf_java_stacks_01_all,
+    perf_java_stacks_01_addrs,
+
+    perf_java_stacks_02_pid,
+    perf_java_stacks_02_tid,
+    perf_java_stacks_02_kernel,
+    perf_java_stacks_02_jit,
+    perf_java_stacks_02_all,
+    perf_java_stacks_02_addrs,
+
+    perf_js_stacks_01_pid,
+    perf_js_stacks_01_tid,
+    perf_js_stacks_01_kernel,
+    perf_js_stacks_01_jit,
+    perf_js_stacks_01_all,
+    perf_js_stacks_01_addrs,
+
+    perf_mirageos_stacks_01_pid,
+    perf_mirageos_stacks_01_tid,
+    perf_mirageos_stacks_01_kernel,
+    perf_mirageos_stacks_01_jit,
+    perf_mirageos_stacks_01_all,
+    perf_mirageos_stacks_01_addrs,
+
+    perf_numa_stacks_01_pid,
+    perf_numa_stacks_01_tid,
+    perf_numa_stacks_01_kernel,
+    perf_numa_stacks_01_jit,
+    perf_numa_stacks_01_all,
+    perf_numa_stacks_01_addrs,
+
+    perf_rust_Yamakaky_dcpu_pid,
+    perf_rust_Yamakaky_dcpu_tid,
+    perf_rust_Yamakaky_dcpu_kernel,
+    perf_rust_Yamakaky_dcpu_jit,
+    perf_rust_Yamakaky_dcpu_all,
+    perf_rust_Yamakaky_dcpu_addrs,
+
+    perf_vertx_stacks_01_pid,
+    perf_vertx_stacks_01_tid,
+    perf_vertx_stacks_01_kernel,
+    perf_vertx_stacks_01_jit,
+    perf_vertx_stacks_01_all,
+    perf_vertx_stacks_01_addrs,
+}
+
+fn test_collapse_perf(test_file: &str, expected_file: &str, options: Options) -> io::Result<()> {
+    let r = BufReader::new(File::open(test_file)?);
+
+    let mut result = Cursor::new(Vec::with_capacity(expected_file.len()));
+    handle_file(options, r, &mut result)?;
+    let mut expected = BufReader::new(File::open(expected_file)?);
+
+    result.set_position(0);
+
+    let mut buf = String::new();
+    for line in result.lines() {
+        // Strip out " and ' since perl version does.
+        let line = line?.replace("\"", "").replace("'", "");
+        expected.read_line(&mut buf)?;
+        assert_eq!(line, buf.trim_end());
```

Comment:
It'd be awesome if this could also include the line number. I guess the file is visible from the test name?

Reply:
Also, I wonder if we might want to use something like [`pretty_assertions`](https://github.com/colin-kiegel/rust-pretty-assertions) to make these things easier to debug.

Reply:
I agree with both suggestions. pretty_assertions looks awesome!